src/core/managers/internal/ToastManager/helpers/tests/processTransactions.test.ts

```diff
@@ -0,0 +1,253 @@
+import {
+  getIsTransactionFailed,
+  getIsTransactionPending,
+  getIsTransactionSuccessful,
+  getIsTransactionTimedOut
+} from 'store/actions/transactions/transactionStateByStatus';
+import { TransactionServerStatusesEnum } from 'types/enums.types';
+import { createTransactionToast } from '../createTransactionToast';
+import { processTransactions } from '../processTransactions';
+import {
+  createMockSessions,
+  createMockToastSlice,
+  createMockTransaction
+} from './mocks/mockTypes';
+import { ITransactionToast } from '../../types/toast.types';
+// Mock dependencies
+jest.mock('core/methods/network/getExplorerAddress', () => ({
+  getExplorerAddress: jest.fn()
+}));
+
+jest.mock('store/actions/transactions/transactionStateByStatus', () => ({
+  getIsTransactionPending: jest.fn(),
+  getIsTransactionTimedOut: jest.fn(),
+  getIsTransactionFailed: jest.fn(),
+  getIsTransactionSuccessful: jest.fn()
+}));
+
+jest.mock('../createTransactionToast', () => ({
+  createTransactionToast: jest.fn()
+}));
+
+describe('processTransactions', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    // Default mock values
+    (getExplorerAddress as jest.Mock).mockReturnValue('https://explorer.test');
+    (getIsTransactionPending as jest.Mock).mockReturnValue(false);
+    (getIsTransactionTimedOut as jest.Mock).mockReturnValue(false);
+    (getIsTransactionFailed as jest.Mock).mockReturnValue(false);
+    (getIsTransactionSuccessful as jest.Mock).mockReturnValue(false);
+    (createTransactionToast as jest.Mock).mockImplementation(({ toast }) => ({
+      toastId: toast.toastId,
+      mockToast: true
+    }));
+  });
+
+  it('should return empty arrays when no transactions exist', () => {
+    // Setup
+    const toastList = createMockToastSlice([]);
+    const sessions = {};
+    const account = { address: 'user-address' };
+
+    // Execute
+    const result = processTransactions(toastList, sessions, account);
+
+    // Verify
+    expect(result).toEqual({
+      processingTransactions: [],
+      completedTransactions: []
+    });
+    expect(getExplorerAddress).toHaveBeenCalled();
+  });
+
+  it('should classify pending transactions correctly', () => {
+    // Setup - mock a pending transaction
+    (getIsTransactionPending as jest.Mock).mockReturnValue(true);
+
+    const toastList = createMockToastSlice(['toast-1']);
+    const sessions = createMockSessions(['toast-1'], 'pending');
+    const account = { address: 'user-address' };
+
+    // Mock the toast creation
+    const mockTransactionToast = { toastId: 'toast-1', mockToast: true };
+    (createTransactionToast as jest.Mock).mockReturnValue(mockTransactionToast);
+
+    // Execute
+    const result = processTransactions(toastList, sessions, account);
+
+    // Verify
+    expect(result.processingTransactions).toContain(mockTransactionToast);
+    expect(result.completedTransactions).toHaveLength(0);
+    expect(createTransactionToast).toHaveBeenCalledWith(
```

Comment:
I find this overwhelming. Why not have simple tests for functions instead of complex ones?